### PR TITLE
Developer guide rewrite

### PR DESCRIPTION
**Pull request by qualidafial**
Work in progress.

See outline at https://docs.google.com/document/d/1opv-OpOQTdYJNo5B1pKFJDrCIVT3vEBiDEQ_6nwgzRw/edit?usp=sharing

Strikethrough in the outline doc means "this is done."

Note that I've inlined all adoc files into a single file for the moment, to make it easier to reason about the overall narrative. Once we're done changing things it will probably be appropriate to extract sections back out to their own files.

To do:

* [ ] Find a different phrase than "you can _"

### CODE REVIEW

**Comment by qualidafial** [comment]
@stevenschlansker @arteam This is incomplete, but should give us a clear way forward to complete our documentation. Would you please review?

**Comment by qualidafial** [comment]
@stevenschlansker @arteam Ping. Any chance you could review this PR? The developer guide isn't complete, but I've littered it with todos that I _believe_ covers all the topics that should be included.

File: docs/src/test/java/jdbi/doc/ResultsTest.java [hunk 1]
```diff
@@ -83,61 +81,67 @@ public void findBob() {
...
+        // end::rowMapper[]
+
+        assertThat(users).hasSize(4);
+        assertThat(users.get(3).name).isEqualTo("Data");
+    }
+
+    @Test
+    public void inlineRowMapper() {
+        // tag::inlineRowMapper[]
+        List<User> users = handle.createQuery("SELECT * FROM user ORDER BY id ASC")
```
Comment (arteam) [inline docs/src/test/java/jdbi/doc/ResultsTest.java]: Maybe it's better to write `SELECT id, name` instead of `SELECT *`

**Comment by stevenschlansker** [comment]
Looking now

**Comment by qualidafial** [comment]
Thanks for the reviews guys. Merging.

**Comment by qualidafial** [comment]
Oh, forgot to address @arteam's comment. Will fix that then merge.

**Pull request merged**